trillian_log_api.proto

```diff
@@ -53,10 +53,9 @@ message LogLeaf {
 }
 
 message Node {
```

Comment:
Now that `Node` only only contains `node_hash`, could we convert 
`repeated Node proof_node` into `repeated bytes proof_node`? 
This would make verifying inclusion proofs easier. 

The current proto requires something like:
```go
proof := make([]byte, 0, len(Proof.ProofNode))
for _, n := range Proof.ProofNode {
   proof = append(proof, n.NodeHash)
}
VerifyInclusionProof(..., proof, ...) 
```

If `Proof.ProofNode` was `repeated bytes` This could be simplified to 
```go
VerifyInclusionProof(..., Proof.ProofNode, ...) 

```